plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergV2.java

```diff
@@ -784,6 +784,31 @@ public void testSnapshotReferenceSystemTable()
                         "('main', 'BRANCH', " + snapshotId3 + ", null, null, null)");
     }
 
+    @Test
```

Comment:
`TestIcebergReadVersionedTable` seems a better class for testing time travel. (No need to change this PR because it requires more work)

Reply:
Sure sounds good, when I follow up with CharType support https://github.com/trinodb/trino/pull/19111#discussion_r1332290919 I can move all the test in a separate commit.